core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt

```diff
@@ -181,23 +245,69 @@ class EventsFileManager(
     }
 
     // write to underlying file
-    private fun writeToFile(content: ByteArray, file: File) {
-        FileOutputStream(file, true).use {
-            it.write(content)
-            it.flush()
+    private fun writeToFile(
+        content: ByteArray,
+        file: File,
+    ) {
+        try {
+            FileOutputStream(file, true).use {
+                it.write(content)
+                it.flush()
+            }
+        } catch (e: FileNotFoundException) {
+            logger.error("File not found: ${file.path}")
+        } catch (e: IOException) {
+            logger.error("Failed to write to file: ${file.path}")
+        } catch (e: SecurityException) {
+            logger.error("Security exception when saving event: ${e.message}")
+        } catch (e: Exception) {
+            logger.error("Failed to write to file: ${file.path}")
         }
     }
 
-    private fun writeToFile(content: String, file: File) {
-        file.createNewFile()
-        FileOutputStream(file).use {
-            it.write(content.toByteArray())
-            it.flush()
+    private fun writeEventsToSplitFile(
+        events: List<JSONObject>,
+        file: File,
+    ) {
+        val contents = events.joinToString(separator = "\n", postfix = "\n") { it.toString() }
+        try {
+            file.createNewFile()
+            writeToFile(contents.toByteArray(), file)
+            file.renameTo(File(directory, file.nameWithoutExtension))
+        } catch (e: IOException) {
+            logger.error("Failed to create or write to split file: ${file.path}")
         }
-        file.renameTo(File(directory, file.nameWithoutExtension))
     }
 
     private fun reset() {
         curFile.remove(storageKey)
     }
+
+    private fun handleV1Files() {
+        if (kvs.getLong(storageVersionKey, 1L) > 1L) {
+            return
+        }
+        val unFinishedFiles =
```

Comment:
Instead of just finishing the files, can we just migrate them over to the new format? That way the read path only needs to worry about the latest format. I'd imagine over time we'd have methods like `handleV1Files`, `handleV2Files` etc and the read path only deals with the latest format

Reply:
I agree, I would prefer a one time migration to checking the format of each file

Reply:
I guess it is just we prefer the io at migration time or read time. Originally thought maybe we could save some effort as we need to handle the read io anyway, though agreed it could be more cleanerfor read once we migrate completely.

Reply:
Migration updated to run at first. Kept the reading for old version files for now just in case anything happened in the migration process.